packages/swapper/src/swappers/JupiterSwapper/swapperApi/getTradeRate.ts

```diff
@@ -139,6 +199,40 @@ export const getTradeRate = async (
     {} as Record<AssetId, ProtocolFee>,
   )
 
+  if (feeAccountInstruction) {
+    if (solAsset) {
+      protocolFees[solAssetId] = {
+        requiresBalance: true,
+        amountCryptoBaseUnit: bnOrZero(protocolFees[solAssetId]?.amountCryptoBaseUnit)
+          .plus(PDA_ACCOUNT_CREATION_COST)
+          .toFixed(),
+        asset: solAsset,
+      }
+    }
+  }
+
+  if (input.sendAddress && solAsset) {
+    const { instruction: createTokenAccountInstruction } = buyAsset
+      ? await adapter.createAssociatedTokenAccountInstruction({
+          from: input.sendAddress,
+          to: receiveAddress ?? input.sendAddress,
+          tokenId: fromAssetId(
+            buyAsset.assetId === solAssetId ? wrappedSolAssetId : buyAsset.assetId,
+          ).assetReference,
+        })
+      : { instruction: undefined }
+
+    if (createTokenAccountInstruction) {
+      protocolFees[solAssetId] = {
+        requiresBalance: true,
+        amountCryptoBaseUnit: bnOrZero(protocolFees[solAssetId]?.amountCryptoBaseUnit)
```

Comment:
same here re: circular ref

Reply:
Changed in [6840b5c](https://github.com/shapeshift/web/pull/8267/commits/6840b5c3ff9eefd79239117bbd0c17cdb1c4bebf)